authenticity-collector/src/identity/identity.controller.spec.ts

```diff
@@ -0,0 +1,87 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { MongoMemoryServer } from 'mongodb-memory-server';
+import { IdentityController } from './identity.controller';
+import { IdentityService } from './identity.service';
+import { Identity, IdentitySchema } from './schemas/identity.schema';
+
+describe('IdentityController', () => {
+  let controller: IdentityController;
+  let identityService: IdentityService;
+  let body;
+  let response: Identity;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [
+        HttpModule,
+        ConfigModule,
+        MongooseModule.forRootAsync({
+          useFactory: async () => {
+            let mongod = await MongoMemoryServer.create();
+            let mongoUri = await mongod.getUri();
+            return {
+              uri: mongoUri
+            }
+          }
+        }),
+        MongooseModule.forFeature([{ name: 'Identity', schema: IdentitySchema }])],
+      controllers: [IdentityController],
+      providers: [IdentityService]
+    }).compile();
+
+    controller = module.get<IdentityController>(IdentityController);
+    identityService = module.get<IdentityService>(IdentityService);
+
+    body = {
+      did: "did:iota:2xCZnoUYakpLYzSWXjwiebYp6RpiUi8DvD9DwoU3qe335sdd",
+      timestamp: "2022-01-27T13:04:18.559Z",
+      signature: "2MrtMZZYmKUrB2jdsG4hwzD6yxAjo3uUrnNq44uVFWd6p8zvaRqhwvfQV5keGdJXV57HS7V9djWM5ZSm8dwY7FNm"
+    }
+    response = {
+      did: "did:iota:2xCZnoUYakpLYzSWXjwiebYp6RpiUi8DvD9DwoU3qe335sdd",
+      timestamp: new Date("2022-01-27T13:04:18.559Z"),
+      signature: "2MrtMZZYmKUrB2jdsG4hwzD6yxAjo3uUrnNq44uVFWd6p8zvaRqhwvfQV5keGdJXV57HS7V9djWM5ZSm8dwY7FNm"
+    }
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
+  });
+
+  it('should return a list of auth proves of a device', async () => {
+    const result: Identity[] = [{
+      did: "did:iota:2xCZnoUYakpLYzSWXjwiebYp6RpiUi8DvD9DwoU3qe335sdd",
+      timestamp: new Date("2022-01-27T13:04:18.559Z"),
+      signature: "2MrtMZZYmKUrB2jdsG4hwzD6yxAjo3uUrnNq44uVFWd6p8zvaRqhwvfQV5keGdJXV57HS7V9djWM5ZSm8dwY7FNm"
+    },
+    {
+      did: "did:iota:3xCZnoUYakpLYzSWXjwiebYp6RpiUi8DvD9DwoU3qe335sdd",
+      timestamp: new Date("2022-02-27T13:04:18.559Z"),
+      signature: "3MrtMZZYmKUrB2jdsG4hwzD6yxAjo3uUrnNq44uVFWd6p8zvaRqhwvfQV5keGdJXV57HS7V9djWM5ZSm8dwY7FNH"
+    }
+  ];
+    jest.spyOn(identityService, 'getAuthProves').mockResolvedValue(result);
+    let did = "did:iota:2xCZnoUYakpLYzSWXjwiebYp6RpiUi8DvD9DwoU3qe335sdd";
+    let from = "2022-01-27T13:04:18.559Z";
+    let to = "2022-02-27T13:04:18.559Z";
+    expect(await controller.getAuthProves({did: did,from: from,to: to})).toBe(result);
+  })
+
+  it('should save identity', async () => {
+    jest.spyOn(identityService, 'proveAndSaveSlu').mockResolvedValue(response);
+    expect(await controller.saveSlu(body)).toBe(response)
```

Comment:
nothing wrong here but imo it would be better to read if you always first call the function and save the response in a local var as following:
```
    const res = await controller.saveSlu(body)
    expect(res).toBe(response)
    ```

please also consider this in the other spec files